### PR TITLE
fix crash when app is killed or closed

### PR DESCRIPTION
**Pull request by diegohzea**
the app currently crash when is killed/closed because de dart executor returns null.

### CODE REVIEW

**Comment by osama450** [comment]
hi @diegohzea , im pub get from your fork and get this error when I try close overly many times only 
java.lang.NullPointerException: Attempt to invoke interface method 'void android.view.WindowManager.updateViewLayout(android.view.View, android.view.ViewGroup$LayoutParams)' on a null object reference
E/AndroidRuntime(29160): at flutter.overlay.window.flutter_overlay_window.OverlayService$TrayAnimationTimerTask.lambda$run$0$flutter-overlay-window-flutter_overlay_window-OverlayService$TrayAnimationTimerTask(OverlayService.java:355)
E/AndroidRuntime(29160): at flutter.overlay.window.flutter_overlay_window.OverlayService$TrayAnimationTimerTask$$ExternalSyntheticLambda0.run(Unknown Source:2)
E/AndroidRuntime(29160): at android.os.Handler.handleCallback(Handler.java:900)
E/AndroidRuntime(29160): at android.os.Handler.dispatchMessage(Handler.java:103)
E/AndroidRuntime(29160): at android.os.Looper.loop(Looper.java:219)
E/AndroidRuntime(29160): at android.app.ActivityThread.main(ActivityThread.java:8668)
E/AndroidRuntime(29160): at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime(29160): at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:513)
E/AndroidRuntime(29160): at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1109)

**Comment by diegohzea** [comment]
> hi @diegohzea , im pub get from your fork and get this error when I try close overlay many times only java.lang.NullPointerException: Attempt to invoke interface method 'void android.view.WindowManager.updateViewLayout(android.view.View, android.view.ViewGroup$LayoutParams)' on a null object reference E/AndroidRuntime(29160): at flutter.overlay.window.flutter_overlay_window.OverlayService$TrayAnimationTimerTask.lambda$run$0$flutter-overlay-window-flutter_overlay_window-OverlayService$TrayAnimationTimerTask(OverlayService.java:355) E/AndroidRuntime(29160): at flutter.overlay.window.flutter_overlay_window.OverlayService$TrayAnimationTimerTask$$ExternalSyntheticLambda0.run(Unknown Source:2) E/AndroidRuntime(29160): at android.os.Handler.handleCallback(Handler.java:900) E/AndroidRuntime(29160): at android.os.Handler.dispatchMessage(Handler.java:103) E/AndroidRuntime(29160): at android.os.Looper.loop(Looper.java:219) E/AndroidRuntime(29160): at android.app.ActivityThread.main(ActivityThread.java:8668) E/AndroidRuntime(29160): at java.lang.reflect.Method.invoke(Native Method) E/AndroidRuntime(29160): at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:513) E/AndroidRuntime(29160): at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1109)

Hello, try again with flutter pub upgrade flutter_overlay_window, I added validation to fix this issue.

**Comment by osama450** [comment]
i get this error when run : 
/Users/osamamohamed/.pub-cache/git/flutter_overlay_window-b39ee3dfbfd95128bf9828714564d37b039aed90/android/src/main/java/flutter/overlay/window/flutter_overlay_window/OverlayService.java:328: error: incompatible types: missing return value
                        if (windowManager == null) return;
                                                   ^
Note: /Users/osamamohamed/.pub-cache/git/flutter_overlay_window-b39ee3dfbfd95128bf9828714564d37b039aed90/android/src/main/java/flutter/overlay/window/flutter_overlay_window/OverlayService.java uses or overrides a deprecated API.

@diegohzea

**Comment by osama450** [comment]
i fixed it by add if (windowManager == null) return false it is run but same crash 

java.lang.NullPointerException: Attempt to invoke interface method 'void android.view.WindowManager.updateViewLayout(android.view.View, android.view.ViewGroup$LayoutParams)' on a null object reference
E/AndroidRuntime(17110): 	at flutter.overlay.window.flutter_overlay_window.OverlayService$TrayAnimationTimerTask.lambda$run$0$flutter-overlay-window-flutter_overlay_window-OverlayService$TrayAnimationTimerTask(OverlayService.java:373)

**Comment by diegohzea** [comment]
> i fixed it by add if (windowManager == null) return false it is run but same crash
> 
> java.lang.NullPointerException: Attempt to invoke interface method 'void android.view.WindowManager.updateViewLayout(android.view.View, android.view.ViewGroup$LayoutParams)' on a null object reference E/AndroidRuntime(17110): at flutter.overlay.window.flutter_overlay_window.OverlayService$TrayAnimationTimerTask.lambda$run$0$flutter-overlay-window-flutter_overlay_window-OverlayService$TrayAnimationTimerTask(OverlayService.java:373)

sorry for that, try again.

File: android/src/main/java/flutter/overlay/window/flutter_overlay_window/OverlayService.java [hunk 1]
```diff
@@ -84,24 +94,13 @@ public void onDestroy() {
...
-            }
-            isRunning = false;
-            return START_STICKY;
-        }
-        if (windowManager != null) {
-            windowManager.removeView(flutterView);
-            windowManager = null;
-            flutterView.detachFromFlutterEngine();
+        if (intent != null && OverlayConstants.ACTION_STOP_OVERLAY.equals(intent.getAction())) {
+            stopForeground(true);
```
Comment (saurabhkumar8112) [inline android/src/main/java/flutter/overlay/window/flutter_overlay_window/OverlayService.java:98]: Is there any reason for detaching the service from the foreground?
Also from API 33, this method has been deprecated so you will have to pass the proper notification flag behaviour
https://developer.android.com/reference/android/app/Service#stopForeground(boolean)

**Comment by Infinity-Studios-Software** [comment]
does this work with API 34?

I'm losing the ability to communicate between the app and the overlay when closing the the app and restarting it, even after dismissing the overlay and restarting it

I started getting this error in the official package

**Comment by X-SLAYER** [comment]
Thank you already fixed in the latest version

**Pull request closed without merging**